functests/2_performance_update/kubelet.go

```diff
@@ -0,0 +1,169 @@
+			for _, node := range workerRTNodes {
+				kubeletConfig, err := nodes.GetKubeletConfig(&node)
+				Expect(err).ToNot(HaveOccurred())
+				sysctlsValue := kubeletConfig.AllowedUnsafeSysctls
+				Expect(sysctlsValue).Should(ContainElements("net.core.somaxconn", "kernel.msg*"))
+				Expect(kubeletConfig.KubeReserved["memory"]).To(Equal("768Mi"))
+				Expect(kubeletConfig.ImageMinimumGCAge.Seconds()).To(Equal(180))
+			}
+			for _, node := range workerRTNodes {
+				stdout, err := nodes.ExecCommandOnNode(kubeletArguments, &node)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(strings.Contains(stdout, "300Mi")).To(BeTrue())
+			}
+		})
+		Context("When setting cpu manager related parameters", func() {
+			It("[test_id:45493]Should not override performance-addon-operator values", func() {
+				cpuManagerAnnotation := map[string]string{
+					"kubeletconfig.experimental": "{\"cpuManagerPolicy\":\"static\",\"cpuManagerReconcilePeriod\":\"5s\"}",
+				}
+				profile.SetAnnotations(cpuManagerAnnotation)
+				By("Applying changes in performance profile and waiting until mcp will start updating")
+				profiles.UpdateWithRetry(profile)
+				mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
+				By("Waiting when mcp finishes updates")
+				mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+				for _, node := range workerRTNodes {
+					kubeletConfig, err := nodes.GetKubeletConfig(&node)
+					Expect(err).ToNot(HaveOccurred())
+					Expect(kubeletConfig.CPUManagerPolicy).Should(Equal("static"))
+					Expect(kubeletConfig.CPUManagerReconcilePeriod.Seconds()).To(Equal(5))
+				}
+			})
+		})
+		It("[test_id:45490]Test memory reservation changes", func() {
+			memoryAnnotation := map[string]string{
+				"kubeletconfig.experimental": "{\"systemReserved\":{\"memory\":\"300Mi\"},\"kubeReserved\":{\"memory\":\"768Mi\"}}",
+			}
+			profile.SetAnnotations(memoryAnnotation)
+			By("Applying changes in performance profile and waiting until mcp will start updating")
+			profiles.UpdateWithRetry(profile)
+			mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
+			By("Waiting when mcp finishes updates")
+			mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+			for _, node := range workerRTNodes {
+				kubeletConfig, err := nodes.GetKubeletConfig(&node)
+				Expect(err).ToNot(HaveOccurred())
+				totalCapactity := node.Status.Capacity.Memory().MilliValue()
+				evictionMemory := kubeletConfig.EvictionHard["memory.available"]
+				kubeReserved := kubeletConfig.KubeReserved["memory"]
+				evictionMemoryInt, err := strconv.ParseInt(strings.TrimSuffix(evictionMemory, "Mi"), 10, 64)
+				kubeReservedMemoryInt, err := strconv.ParseInt(strings.TrimSuffix(kubeReserved, "Mi"), 10, 64)
+				systemReservedResource := resource.NewQuantity(300*1024*1024, resource.BinarySI)
```

Comment:
maybe we should add a comment explaining these values (hard coded for a reason ?) or calculation, its a bit hard to understand for a first time reader.  just a suggestion, not mandatory.

Reply:
okay , will put the comments

Reply:
I will post a new patch for this, with comments, and probably rewrite this.

Reply:
@yanirq @fromanirh i have added comments, Hope this makes the test case clear